apps/spark/src/main/java/com/linkedin/openhouse/jobs/scheduler/JobsScheduler.java

```diff
@@ -124,11 +124,14 @@ public static void main(String[] args) {
                 OperationTask.POLL_INTERVAL_MS_DEFAULT),
             NumberUtils.toLong(
                 cmdLine.getOptionValue("taskTimeoutMs"), OperationTask.TIMEOUT_MS_DEFAULT));
-    JobsScheduler app =
-        new JobsScheduler(
-            Executors.newFixedThreadPool(getNumParallelJobs(cmdLine)),
-            tasksFactory,
-            tablesClientFactory.create());
+    ThreadPoolExecutor executors =
+        new ThreadPoolExecutor(
+            getNumParallelJobs(cmdLine),
+            getNumParallelJobs(cmdLine),
+            0L,
```

Comment:
I think we should have some higher value for keep live time (60 sec or more) as it determines how long a thread can be idle before terminating.

Reply:
We can adjust it to higher value when we switch to batch discovery and submission. For now 0L should be good as default ExecutorSevice default implementation with fixedThreadPool internally uses 0L and creates ThreadPoolExecutor. If shutdownNow on executorService works then we probably don't need to explicitly create this.

Reply:
`keepAliveTime` doesn't matter when `core threads` == `max threads`.